--- spack/lib/spack/spack/fetch_strategy.py.orig	2020-12-16 15:40:27.953050684 +0000
+++ spack/lib/spack/spack/fetch_strategy.py	2020-12-16 17:59:13.538267644 +0000
@@ -301,6 +301,11 @@
         url = None
         errors = []
         for url in self.candidate_urls:
+            if url[0:5] == 'azure':
+               import spack.util.azure_blob as azure_blob_util
+               parsed_url = urllib_parse.urlparse(url)
+               azureblob = azure_blob_util.AzureBlob(parsed_url)
+               url = azureblob.azure_url_sas()
             if not self._existing_url(url):
                 continue
 
@@ -1361,6 +1366,7 @@
             'https': 'url',
             'ftp': 'url',
             'ftps': 'url',
+            'azure': 'url',
         })
 
     scheme = parsed_url.scheme
