--- spack/lib/spack/spack/fetch_strategy.py_orig	2020-08-31 19:13:32.089112096 +0000
+++ spack/lib/spack/spack/fetch_strategy.py	2020-08-31 19:15:52.096434168 +0000
@@ -298,6 +298,11 @@
         errors = []
         for url in self.candidate_urls:
             try:
+                if url[0:5] == 'azure':
+                   import spack.util.azure_blob as azure_blob_util
+                   parsed_url = urllib_parse.urlparse(url)
+                   azureblob = azure_blob_util.AzureBlob(parsed_url)
+                   url = azureblob.azure_url_sas()
                 partial_file, save_file = self._fetch_from_url(url)
                 if save_file:
                     os.rename(partial_file, save_file)
@@ -1347,6 +1352,7 @@
             'https': 'url',
             'ftp': 'url',
             'ftps': 'url',
+            'azure': 'url',
         })
 
     scheme = parsed_url.scheme
